angular-ngrx-scss/src/app/state/profile/profile.effects.ts

```diff
@@ -16,12 +18,54 @@ export class ProfileEffects {
       switchMap(({ username }) =>
         combineLatest([
           this.userService.getUserInfo(username),
-          this.userService.getUserOrganizations(),
+          this.userService.getUserOrganizations(username),
           this.userService.getUserRepos(username),
         ]).pipe(
-          map(([user, orgs, repos]) =>
-            fetchProfileSuccess({ data: { user, orgs, repos } }),
-          ),
+          map(([userData, orgsData, reposData]) => {
+            const user: UserState = {
+              avatar: userData.avatar_url,
+              bio: userData.bio,
+              blog: userData.blog,
+              company: userData.company,
+              email: userData.email,
+              followers: userData.followers,
+              following: userData.following,
+              location: userData.location,
+              name: userData.name,
+              twitter_username: userData.twitter_username,
+              username: userData.login,
+              type: userData.type,
+            };
+            const orgs: UserOrgsState[] = orgsData.map((org) => ({
+              id: org.id,
+              login: org.login,
+              avatar_url: org.avatar_url,
+            }));
```

Comment:
I think it would make this code much more readable if we extracted the mapping functions instead of inlining them here

Reply:
have you created a ticket for using mapping functions? AFAIR we discussed putting a POC in this PR and then following up for other places in another PR

Reply:
Ticket: https://github.com/thisdot/starter.dev-github-showcases/issues/825